### ISSUE TITLE
Incorrect umask check handle for multiple files

### ISSUE THREAD

**Issue by gkrystev** [CISOfy/lynis#559]
In case when umask is checked in multiple files and in some of the files except the last one a weak umask is found, the tool reports weak mask for the rest of the files. In the example bellow, the weak umask is only in /etc/csh.cshrc. However, the check /etc/profile is reported weak as well.

Expected:
  - Checking default umask values
    - Checking default umask in /etc/bashrc                   [ OK ]
    - Checking default umask in /etc/csh.cshrc                [ WEAK ]
    - Checking default umask in /etc/profile                  [ OK ]

Actual:
  - Checking default umask values
    - Checking default umask in /etc/bashrc                   [ OK ]
    - Chhhhecking default umask in /etc/csh.cshrc             [ WEAK ]
    - Chhhhecking default umask in /etc/profile               [ WEAK ]

The issue comes from file "include/tests_shells" where in line 238, the variable HARDENING_POSSIBLE is outside the loop for different files check. When the variable is set to 1, for the next file the value is not reset to 0 in order to pass the check.

Current code:
```
    232     # Test        : SHLL-6230
    233     # Description : Check for umask values in shell configurations
    234     SHELL_CONFIG_FILES="${ROOTDIR}etc/bashrc ${ROOTDIR}etc/bash.bashrc ${ROOTDIR}etc/csh.cshrc ${ROOTDIR}etc/profile"
    235     Register --test-no SHLL-6230 --weight H --network NO --category security --description "Perform umask check for shell configurations"
    236     if [ ${SKIPTEST} -eq 0 ]; then
    237         FOUND=0
    238         HARDENING_POSSIBLE=0
    239         Display --indent 2 --text "- Checking default umask values"
    240         for FILE in ${SHELL_CONFIG_FILES}; do
    241             FIND=""
    242             if [ -f ${FILE} ]; then
    243                 LogText "Result: file ${FILE} exists"
```

Suggested code:
```
    232     # Test        : SHLL-6230
    233     # Description : Check for umask values in shell configurations
    234     SHELL_CONFIG_FILES="${ROOTDIR}etc/bashrc ${ROOTDIR}etc/bash.bashrc ${ROOTDIR}etc/csh.cshrc ${ROOTDIR}etc/profile"
    235     Register --test-no SHLL-6230 --weight H --network NO --category security --description "Perform umask check for shell configurations"
    236     if [ ${SKIPTEST} -eq 0 ]; then
    237         FOUND=0
    238         Display --indent 2 --text "- Checking default umask values"
    239         for FILE in ${SHELL_CONFIG_FILES}; do
    240             FIND=""
    241             HARDENING_POSSIBLE=0
    242             if [ -f ${FILE} ]; then
    243                 LogText "Result: file ${FILE} exists"
```

**Comment by mboelen** [comment]
Sounds like a good proposal. Can you create a pull request for it?

**Comment by mboelen** [comment]
Fix was applied via #560
